includes/sitealias.inc

```diff
@@ -1705,8 +1707,8 @@ function _drush_sitealias_set_context_by_name($alias, $prefix = '') {
       drush_sitealias_cache_alias_by_path($site_alias_settings);
       if (empty($prefix)) {
         // Create an alias '@self'
-        _drush_sitealias_cache_alias('@self', $site_alias_settings);
-        // change the selected site to match the new --root and --uri, if any were set
+        drush_sitealias_create_self_alias();
```

Comment:
drush_sitealias_create_self_alias() is preferred here since it considers the --uri override provided via CLI as opposed to  _drush_sitealias_cache_alias('@self', $site_alias_settings) which appears not to.